consumer-java/aws-kinesis-video-producer-sdk-canary-consumer/src/main/java/com/amazon/kinesis/video/canary/consumer/ProducerSdkCanaryConsumer.java

```diff
@@ -60,18 +58,14 @@ public void process(InputStream inputStream, FragmentMetadataCallback fragmentMe
                 credentialsProvider, streamName, new StartSelector().withStartSelectorType(StartSelectorType.NOW),
                 amazonKinesisVideo,
                 consumerFactory);
-        getMediaWorker.run();
-    }
 
-    private static String getCanaryStr(int canaryType) {
-        switch (canaryType) {
-            case 0:
-                return "realtime";
-            case 1:
-                return "offline";
-            default:
-                return "";
-        }
+        Terminate terminate = new Terminate();
```

Comment:
Not sure why we use raw threads here. Let’s see if we can introduce an executor to run a runnable or a callable

Reply:
Let me look into that. My java knowledge is not great. So would be happy to look into alternatives